src/Mod/Path/PathScripts/PathDressupZCorrect.py

```diff
@@ -0,0 +1,342 @@
+# *   Copyright (c) 2013 by Panagiotis Mavrogiorgos                         *
+# *                                                                         *
+# ***************************************************************************
+import FreeCAD
+import FreeCADGui
+import Part
+import Path
+import PathScripts.PathGeom as PathGeom
+import PathScripts.PathLog as PathLog
+import PathScripts.PathUtils as PathUtils
+#from bisect import bisect_left
+
+from PySide import QtCore, QtGui
+
+"""Z Depth Correction Dressup.  This dressup takes a probe file as input and does bilinear interpolation of the Zdepths to correct for a surface which is not parallel to the milling table/bed.  The probe file should conform to the format specified by the linuxcnc G38 probe logging: 9-number coordinate consisting of XYZABCUVW http://linuxcnc.org/docs/html/gcode/g-code.html#gcode:g38
+"""
+
+LOG_MODULE = PathLog.thisModule()
+
+if True:
+    PathLog.setLevel(PathLog.Level.DEBUG, LOG_MODULE)
+    PathLog.setLevel(PathLog.Level.DEBUG, LOG_MODULE)
+else:
+    PathLog.setLevel(PathLog.Level.NOTICE, LOG_MODULE)
+
+
+# Qt tanslation handling
+def translate(context, text, disambig=None):
+    return QtCore.QCoreApplication.translate(context, text, disambig)
+
+movecommands = ['G1', 'G01', 'G2', 'G02', 'G3', 'G03']
+rapidcommands = ['G0', 'G00']
+arccommands = ['G2', 'G3', 'G02', 'G03']
+
+class ObjectDressup:
+
+    def __init__(self, obj):
+        obj.addProperty("App::PropertyLink", "Base", "Path", QtCore.QT_TRANSLATE_NOOP("Path_DressupAxisMap", "The base path to modify"))
+        obj.addProperty("App::PropertyFile", "probefile", "ProbeData", QtCore.QT_TRANSLATE_NOOP("Path_DressupZCorrect", "The point file from the surface probing."))
+        obj.Proxy = self
+        obj.addProperty("Part::PropertyPartShape", "interpSurface", "Path")
+        obj.setEditorMode('interpSurface', 2)  # hide
+        obj.addProperty("App::PropertyDistance", "ArcInterpolate", "Interpolate", QtCore.QT_TRANSLATE_NOOP("Path_DressupZCorrect", "Deflection distance for arc interpolation"))
+        obj.addProperty("App::PropertyDistance", "SegInterpolate", "Interpolate", QtCore.QT_TRANSLATE_NOOP("Path_DressupZCorrectp", "break segments into smaller segments of this length."))
+        obj.ArcInterpolate = 0.1
+        obj.SegInterpolate = 1.0
+    def __getstate__(self):
+        return None
+
+    def __setstate__(self, state):
+        return None
+
+    def onChanged(self, fp, prop):
+        if str(prop) == "probefile":
+            self._loadFile(fp, fp.probefile)
+
+    def _bilinearInterpolate(self, surface, x, y):
+
+        p1 = FreeCAD.Vector(x, y, 100.0)
+        p2 = FreeCAD.Vector(x, y, -100.0)
+
+        vertical_line = Part.Line(p1, p2)
+        points, curves = vertical_line.intersectCS(surface)
+        return points[0].Z
+
+    def _loadFile(self, obj, filename):
+        if filename == "":
+            return
+
+	f1 = open(filename, 'r')
```

Comment:
The Travis tests fail here because there is a Tab, and is inconsistent with the rest of the code, which is indented with spaces.

Reply:
fixed